game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/AaDefenseCalculator.java

```diff
@@ -0,0 +1,96 @@
+package games.strategy.triplea.delegate.power.calculator;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.attachments.UnitSupportAttachment;
+import java.util.function.Predicate;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Value;
+
+/**
+ * Calculates defense strength and roll for AA/Targeted dice
+ *
+ * <p>This takes into account friendly support, and enemy support
+ */
+@Builder
+@Value
+@Getter(AccessLevel.NONE)
+class AaDefenseCalculator implements OffenseOrDefenseCalculator {
```

Comment:
We may want to think about the interface naming. 'OffenseOrDefenseCalculator' seems to be a bit too vague, not necessarily single purpose.

IF you saw for example an instance of the interface passed, `OffenseOrDefenseCalculator`, would its API usage and methods be relatively obvious?

Reply:
One suggestion, we may want to avoid the 'calculator' suffix. That tends to implies relatively stateless objects that are functional. Calculators tend to receive in put values and compute output values, they're essentially functions. It looks like the instances each have a pretty decent amount of state behind them.

Perhaps this class could be called `AaDefenseCombatValue` and the interface `CombatValue`